scripts/process/process_map.R

```diff
@@ -34,7 +35,53 @@ process.state_map <- function(viz){
   hawaii <- elide(hawaii, shift=c(5400000, -1400000))
   row.names(hawaii) <- 'hawaii'
   proj4string(hawaii) <- proj4string(conus)
+  states.out <- rbind(conus, alaska, hawaii, makeUniqueIDs = TRUE)
+  
+  # shifting states to be located at (0,0) in svg-space:
+  shifted.states <- zero_shift_sp(states.out)
+  shifted.centroids <- zero_shift_sp(shifted.states)
+  state.centroids <- state_centroids(states.out) # keeps plot order
+  
+  out <- list(states = states.out, 
+              shifted.states = shifted.states, 
+              shifted.centroids = shifted.centroids,
+              state.centroids = state.centroids, 
+              bbox = bbox(states.out)) # then spatial points for centroids, and other things
   
-  out <- rbind(conus, alaska, hawaii, makeUniqueIDs = TRUE)
   saveRDS(out, file = viz[['location']])
+}
+
+library(rgeos)
+scale_transform <- function(obj, scale){
+  centroid <- c(unlist(rgeos::gCentroid(obj,byid=TRUE)@coords))
+  obj <- elide(obj, scale=max(apply(bbox(obj), 1, diff)) * scale) # do by area?
+  elide(obj, shift=centroid - c(unlist(rgeos::gCentroid(obj,byid=TRUE)@coords)))
+}
+
+state_centroids <- function(sp){
+  state.centroids <- NULL
+  ordered.names <- names(sp)[sp@plotOrder]
+  for (name in ordered.names){
+    obj <- rgeos::gCentroid(sp[names(sp) == name,],byid=TRUE)
+    if (name == ordered.names[1]){
+      state.centroids <- obj
```

Comment:
if someone has a smarter solution for binding an empty sp list, I'm ears. Didn't look very hard and did this instead